csrc/host_ir/lower_to_llvm.cpp

```diff
@@ -0,0 +1,912 @@
+  for (auto input_tv : input_tvs) {
+    input_logical_domains.insert(
+        input_logical_domains.end(),
+        input_tv->getLogicalDomain().begin(),
+        input_tv->getLogicalDomain().end());
+  }
+
+  auto output_logical_domain = output_tv->getLogicalDomain();
+  auto output_allocation_domain = output_tv->getMaybeAllocationDomain();
+
+  auto TSM_logical_shape =
+      generate_infer_shape_module(input_logical_domains, output_logical_domain, *fusion, "infer_logical_shape_module");
+  if (auto Err = pimpl_->jit->addIRModule(std::move(TSM_logical_shape))) {
+    llvm::errs() << "Error adding shape infer module to JIT: "
+                 << llvm::toString(std::move(Err)) << "\n";
+  }
+
+  // JIT compile stride inference module
+  auto TSM_logical_stride =
+      generate_infer_stride_module(output_logical_domain, output_allocation_domain, *fusion, "infer_logical_stride_module");
+  if (auto Err = pimpl_->jit->addIRModule(std::move(TSM_logical_stride))) {
+    llvm::errs() << "Error adding stride infer module to JIT: "
+                 << llvm::toString(std::move(Err)) << "\n";
+  }
+  // Look up the function pointers and store them
+  pimpl_->logical_shape_infer_fn =
+      ExitOnErr(pimpl_->jit->lookup("infer_shape")).toPtr<ShapeInferFunc>();
+  pimpl_->logical_stride_infer_fn =
+      ExitOnErr(pimpl_->jit->lookup("infer_stride")).toPtr<StrideInferFunc>();
+}
+
+at::Tensor HostIrLlvmJit::allocateOutputTensor(const std::vector<at::Tensor>& input_tensors) {
+  NVF_ERROR(
+      pimpl_->logical_shape_infer_fn != nullptr && pimpl_->logical_stride_infer_fn != nullptr
+      && pimpl_->output_tv != nullptr,
+      "JIT must be compiled before running.");
+
+  // Allocate memory for shape result
+  std::vector<int64_t> logical_shape_result(pimpl_->output_tv->getLogicalDomain().size());
+  std::vector<int64_t> input_sizes;
+  for(auto& input_tensor : input_tensors){
+    input_sizes.insert(input_sizes.end(), input_tensor.sizes().begin(), input_tensor.sizes().end());
+  }
+
+  // Run output tensor logical shape inference
+  pimpl_->logical_shape_infer_fn(
+      input_sizes.data(),
+      input_sizes.size(),
+      logical_shape_result.data(),
+      logical_shape_result.size());
+
+  // Allocate memory for stride result
+  std::vector<int64_t> logical_stride_result(pimpl_->output_tv->getLogicalDomain().size());
+  std::vector<int64_t> logical_sharded_shape_result(pimpl_->output_tv->getLogicalDomain().size());
+
+  // Run output tensor logical stride inference
+  pimpl_->logical_stride_infer_fn(
+      logical_shape_result.data(),
+      logical_shape_result.size(),
+      logical_stride_result.data(),
+      logical_stride_result.size(),
+      logical_sharded_shape_result.data(),
+      logical_sharded_shape_result.size());
+
+  // Create the output tensor with the computed shape and strides
+  at::Tensor allocated_tensor = at::empty_strided(logical_sharded_shape_result, logical_stride_result, input_tensors[0].options());
```

Comment:
This is far from what I had in mind. I expect LLVM JIT to take a TensorView* and generate a LLVM IR function like the following

```
at::Tensor allocate_tv1(int64_t i0, int64_t i1, ...) {  // args represent logical domain extents of tv1
  calculate local tensor sizes (e.g. for multi-GPU)
  calculate strides
  return at::empty_strided(local_tensor_sizes, strides) 
}
```

logical_shape_infer_fn and logical_stride_infer_fn shouldn't even exist. They should be inlined to the above `allocate_tv1` function for performance and for simplicity. at::empty_strided should be called by `allocate_tv1` rather than the JIT. 

I'm less worried about this particular IR than what we'll try to deliver by the end of the internship. I'm happy to discuss this in person.